test/metabase/api/card_test.clj

```diff
@@ -674,19 +674,26 @@
                                        :dataset_query (:dataset_query card)})))))
       (testing "You can create a card with a saved question CTE as a model"
         (mt/with-model-cleanup [:model/Card]
-          (let [card-tag-name (str "#" (u/the-id card))]
-            (is (=? {:id pos-int?}
-                    (mt/user-http-request :rasta :post 200 "card"
-                                          (merge
-                                           (mt/with-temp-defaults :model/Card)
-                                           {:dataset_query {:database (u/the-id db)
-                                                            :type     :native
-                                                            :native   {:query         (format "SELECT * FROM {{%s}};" card-tag-name)
-                                                                       :template-tags {card-tag-name {:card-id      (u/the-id card),
-                                                                                                      :display-name card-tag-name,
-                                                                                                      :id           (str (random-uuid))
-                                                                                                      :name         card-tag-name,
-                                                                                                      :type         :card}}}}}))))))))))
+          (let [card-tag-name (str "#" (u/the-id card))
+                dataset-query {:dataset_query {:database (u/the-id db)
+                                               :type     :native
+                                               :native   {:query         (format "SELECT * FROM {{%s}};" card-tag-name)
+                                                          :template-tags {card-tag-name {:card-id      (u/the-id card),
+                                                                                         :display-name card-tag-name,
+                                                                                         :id           (str (random-uuid))
+                                                                                         :name         card-tag-name,
+                                                                                         :type         :card}}}}}
+                {card-id :id
+                 :as     created} (mt/user-http-request :rasta :post 200 "card"
+                                              (merge
+                                                (mt/with-temp-defaults :model/Card)
+                                                dataset-query))
+                retrieved     (mt/user-http-request :rasta :get 200 (str "card/" card-id))]
+            (is (pos-int? card-id))
+            (testing "A POST returns the newly created object, so no follow-on GET is required"
```

Comment:
This PR addresses a bug right? I've seen #ISSUENUM put in testing strings as a reference before, if you want to add that here.

Reply:
Hmm, more of a performance improvement, but i like the idea. Will update.